test/unit/TokensTest.php

```diff
@@ -619,4 +619,21 @@ public function testObtainCachedToken()
         $cachedTokenDetails = $mockClient->obtainNewToken('id', 'secret');
         $this->assertEquals($cachedTokenDetails, $tokenDetails);
     }
+
+    public function testInvalidUserAgentThrowsException()
```

Comment:
Worth adding a test to make sure a valid one gets through plus this doubles up as usage documentation.